framework/src/modules/interoperability/sidechain/commands/mainchain_registration.ts

```diff
@@ -150,37 +180,15 @@ export class MainchainRegistrationCommand extends BaseInteroperabilityCommand {
 				validatorsHash: computeValidatorsHash(mainchainValidators, BigInt(THRESHOLD_MAINCHAIN)),
 			},
 			status: CHAIN_REGISTERED,
-		});
+		};
+		await chainSubstore.set(context, MAINCHAIN_ID_BUFFER, mainchainAccount);
 
 		const channelSubstore = this.stores.get(ChannelDataStore);
 		await channelSubstore.set(context, MAINCHAIN_ID_BUFFER, {
 			inbox: { root: EMPTY_HASH, appendPath: [], size: 0 },
 			outbox: { root: EMPTY_HASH, appendPath: [], size: 0 },
 			partnerChainOutboxRoot: EMPTY_HASH,
-			messageFeeTokenID: { chainID: MAINCHAIN_ID_BUFFER, localID: utils.intToBuffer(0, 4) },
-		});
-
-		const interoperabilityStore = this.getInteroperabilityStore(context);
-
-		const encodedParams = codec.encode(registrationCCMParamsSchema, {
-			networkID: MAINCHAIN_NETWORK_ID,
-			name: MAINCHAIN_NAME,
-			messageFeeTokenID: { chainID: MAINCHAIN_ID_BUFFER, localID: utils.intToBuffer(0, 4) },
-		});
-
-		await interoperabilityStore.sendInternal({
-			module: MODULE_NAME_INTEROPERABILITY,
-			crossChainCommand: CROSS_CHAIN_COMMAND_NAME_REGISTRATION,
-			receivingChainID: MAINCHAIN_ID_BUFFER,
-			fee: BigInt(0),
-			status: CCM_STATUS_OK,
-			params: encodedParams,
-			eventQueue: context.eventQueue,
-			feeAddress: EMPTY_FEE_ADDRESS,
-			getMethodContext: context.getMethodContext,
-			getStore: context.getStore,
-			logger: context.logger,
-			chainID: context.chainID,
+			messageFeeTokenID: TOKEN_ID_LSK_MAINCHAIN,
```

Comment:
`TOKEN_ID_LSK_MAINCHAIN` should be `TOKEN_ID_LSK`?

Reply:
`messageFeeTokenID` has type
export interface MessageFeeTokenID {
	chainID: Buffer;
	localID: Buffer;
}

Reply:
ah yea, i will create separate issue to update this 👍